crates/fluvio-controlplane-metadata/src/smartmodule/spec.rs

```diff
@@ -104,6 +106,40 @@ pub struct SmartModuleWasmSummary {
     pub wasm_length: u32,
 }
 
+pub struct WasmBytes(Vec<u8>);
+
+impl From<Vec<u8>> for WasmBytes {
+    fn from(bytes: Vec<u8>) -> Self {
+        Self(bytes)
+    }
+}
+
+impl Encoder for WasmBytes {
+    fn write_size(&self, _version: Version) -> usize {
+        1
+    }
+
+    fn encode<T>(&self, dest: &mut T, _version: Version) -> Result<(), IoError>
+    where
+        T: BufMut,
+    {
+        if dest.remaining_mut() < 1 {
+            return Err(IoError::new(
+                ErrorKind::UnexpectedEof,
+                "Not enough capacity for WasmBytes",
+            ));
+        }
+
+        dest.put_slice(self.0.as_slice());
```

Comment:
Instead of walking through byte to byte. If the `Encoder` is running on a `WasmBytes` instance, the underlying vector is copied as a slice.